tensorboard/webapp/app_routing/internal_utils.ts

```diff
@@ -24,6 +25,41 @@ import {
   SerializableQueryParams,
 } from './types';
 
+/**
+ * Parses single experiments and short experiment name to id map encoded in
+ * string for card route.
+ *
+ * e.g.
+ *
+ * /card/12345/tag/tagid results,
+ * [
+ *   {name: '', id: '12345'}
+ * ]
+ *
+ * /card/exp1:12345,exp2:99999/tag/tagid results,
+ * [
+ *   {name: 'exp1', id: '12345'},
+ *   {name: 'exp2', id: '99999'}
+ * ]
+ */
+export function parseCardExperimentStr(
+  eidStr: string
+): Array<{name: string; id: string}> {
+  return eidStr.split(',').map((nameToIdStr) => {
+    const colonIndex = nameToIdStr.indexOf(':');
+    if (colonIndex < 0) {
+      return {name: '', id: nameToIdStr};
+    }
```

Comment:
Can we move this part which checks single experiment ahead and then reuse parseCompareExperimentStr for the rest of the code?

Reply:
By reusing parseCompareExperimentStr, do you mean creating a new function with the logic that it uses to check non-single experiments?

Reply:
I mean reuse the parseCompareExperimentStr function below, maybe sth like

```
function parseCardExperimentStr(...) {
    // check if it's single experiment
   ...do some single experiment check and return
   // else it's multiple experiments
   return  parseCompareExperimentStr(...)
}
```

Reply:
Got it!